plugins/arElasticSearchPlugin/lib/model/arElasticSearchAccession.class.php

```diff
@@ -135,39 +135,43 @@ public static function getAccessionEvents($accessionId)
                  event.id,
                  event.type_id,
                  event.date,
-                 property.id AS property_id
+                 note.id AS note_id
                FROM '.QubitAccessionEvent::TABLE_NAME.' event
-               INNER JOIN '.QubitProperty::TABLE_NAME.' property
-               ON (event.id=property.object_id AND property.name="accessionEventNote")
+               INNER JOIN '.QubitNote::TABLE_NAME.' note
+               ON (event.id=note.object_id AND note.type_id=?)
                WHERE event.accession_id = ?';
 
       self::$statements['event'] = self::$conn->prepare($sql);
     }
 
-    self::$statements['event']->execute(array($accessionId));
+    self::$statements['event']->execute([QubitTerm::ACCESSION_EVENT_NOTE_ID, $accessionId]);
 
-    $events = array();
+    $events = [];
     foreach (self::$statements['event']->fetchAll(PDO::FETCH_ASSOC) as $item)
     {
-      $event = array(
-        'id' => $item['id'],
-        'typeId' => $item['type_id'],
-        'date' => $item['date'],
-        'dateString' => $item['date']);
+      $event = ['id' => $item['id']];
 
-      $i18nFields = [];
+      $event['date'] = arElasticSearchPluginUtil::normalizeDateWithoutMonthOrDay($item['date']);
+      $event['dateString'] = Qubit::renderDate($item['date']);
 
-      // Serialize event type term's i18n data
-      $i18nFields = arElasticSearchModelBase::serializeI18ns($item['type_id'], array('QubitTerm'));
+      // Serialize type term
+      if ($item['type_id'])
+      {
+        $node = new arElasticSearchTermPdo($item['type_id']);
+        $serialized = $node->serialize();
+        $event['type'] = ['i18n' => $serialized['i18n']];
```

Comment:
I think this could add other fields apart from the name and there is no need to serialize the entire term. Now that you're using partial foreign types, you could serialize only the required field with ...

```php
$event['type'] = ['i18n' => arElasticSearchModelBase::serializeI18ns(
  $item['type_id'],
  array('QubitTerm'),
  array('fields' => array('name'))
)];
```